js/src/util/wallet.js

```diff
@@ -41,27 +43,26 @@ export class Wallet {
       if (kdfparams.prf !== 'hmac-sha256') {
         throw new Error('Unsupported parameters to PBKDF2');
       }
-      derivedKey = crypto.pbkdf2Sync(pwd, salt, kdfparams.c, kdfparams.dklen, 'sha256');
+      derivedKey = pbkdf2Sync(pwd, salt, kdfparams.c, kdfparams.dklen, 'sha256');
```

Comment:
There is `crypto.pbkdf2` (which is async). Have you checked if the UI still blocks with that one? (It could still be one expensive block even if async, but it sounds promising.)